common/djangoapps/third_party_auth/models.py

```diff
@@ -827,6 +848,77 @@ def get_setting(self, name):
+
+            # When signal handlers are disabled, check if we have a newer configuration
+            latest_config = self._get_latest_configuration()
+            if latest_config and latest_config.id != self.saml_configuration.id:
+                set_custom_attribute('saml_config.using', 'latest_found:id=' + str(latest_config.id))
+                set_custom_attribute('saml_config.outdated_reference', 'true:old_id=' + str(self.saml_configuration.id))
+                return latest_config
+            else:
+                # Either direct reference is still current, or latest lookup failed - use direct reference
+                if latest_config:
+                    set_custom_attribute('saml_config.using', 'direct:id=' + str(self.saml_configuration.id))
+                else:
+                    # Latest lookup failed, but we have a direct reference - use it with warning
+                    set_custom_attribute('saml_config.using', 'direct_fallback:id=' + str(self.saml_configuration.id))
+                    set_custom_attribute('saml_config.latest_lookup_failed', 'true')
+                return self.saml_configuration
+
+        # Fall back to default configuration
+        return self._get_default_configuration()
+
+    def _get_latest_configuration(self):
+        """Get the latest configuration for this provider's site and slug."""
+        # Protect against None saml_configuration
+        if not self.saml_configuration:
+            set_custom_attribute('saml_config.latest_lookup_error', 'no_saml_configuration')
+            return None
+
+        try:
+            return SAMLConfiguration.current(self.site_id, self.saml_configuration.slug)
+        except Exception as e:  # pylint: disable=broad-except
+            # Handle any database, configuration, or attribute errors
+            error_type = type(e).__name__
+            set_custom_attribute('saml_config.latest_lookup_error', error_type + ':' + str(e))
+            return None
+
+    def _get_default_configuration(self):
+        """Get the default configuration, with observability."""
+        try:
+            # SAMLConfiguration.current() returns None if no config found, doesn't raise exceptions
+            # Make sure to use the provider's site_id for proper isolation
+            default_config = SAMLConfiguration.current(self.site_id, 'default')
+            if default_config and default_config.id:  # Ensure it's a valid saved object
+                set_custom_attribute('saml_config.using', 'default:id=' + str(default_config.id))
+                return default_config
+
+            # No valid configuration found
+            set_custom_attribute('saml_config.using', 'none_found')
+            return None
+        except Exception as e:  # pylint: disable=broad-except
+            # Handle any unexpected errors in default configuration lookup
+            set_custom_attribute('saml_config.default_lookup_error', 'error:' + str(e))
+            return None
```

Comment:
I think we are adding a lot of unnecessary complexity. It may not even make sense to have a new function at all. The legacy code already uses `SAMLConfiguration.current(self.site.id, 'default')` on its own. We should either continue to do the same, or if you wish to wrap this single call in the function `_get_default_configuration()` to give it a name, that is acceptable. But why do we need anything else?
```suggestion
        """Get the default configuration."""
            return SAMLConfiguration.current(self.site_id, 'default')
```